modules/gapi/include/opencv2/gapi/cpu/video.hpp

```diff
@@ -8,6 +8,10 @@
 #define OPENCV_GAPI_CPU_VIDEO_API_HPP
 
 #include <opencv2/gapi/gkernel.hpp> // GKernelPackage
+#include "opencv2/core/mat.hpp"
+#include "opencv2/core.hpp"
+
+#undef countNonZero
```

Comment:
Why is this in public GAPI headers?

Reply:
It is banned in tests only.
No idea how this appears in common public header.

Reply:
This appear because both headers `modules/gapi/include/opencv2/gapi/cpu/video.hpp` and `modules/gapi/test/test_precomp.hpp` are included to one `modules/gapi/test/cpu/gapi_video_tests_cpu.cpp`. So, 
`#define countNonZero() countNonZero_is_forbidden_in_tests_use_norm_instead()` in `test_precomp.hpp` affects content of cpu/video.hpp file.

Reply:
Do you really need such details in the future?

Reply:
Please move this hack to the tests if it is really necessary (or move code implementation into `src/`, or replace this call)

`cv::countNonZero(...) > 0)` is slow (and may cause unexpected result overflow which is worse). Replace to
```
norm(..., NORM_INF) != 0
```

Reply:
Ok. Changed.